src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -108,4 +129,104 @@ private void addListingToBusinesses(PlacesSearchResult place) {
+          .location(latLng).radius(50000);
+      try {                                       
+        PlacesSearchResult[] similarBusinessesInTheArea = 
+          request.await().results;
+        if (similarBusinessesInTheArea.length > 1){
+          checkBusinessThroughLinkedin(currentBusiness, similarBusinessesInTheArea);
+        }else if(similarBusinessesInTheArea.length == 1){
+          numberOfSmallBusinesses++;
+        }
+      } catch(GeneralSecurityException | IOException | InterruptedException | ApiException e ) {
+          LOGGER.warning(e.getMessage());
+        } 
+    }
+  }
+  
+  private void checkBusinessThroughLinkedin(Listing currentBusiness, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) 
+                              throws GeneralSecurityException, IOException {
+    String cx = dotenv.get("CX"); 
+    Customsearch cs = new Customsearch.Builder(
+        GoogleNetHttpTransport.newTrustedTransport(), 
+        JacksonFactory.getDefaultInstance(), null) 
+              .setApplicationName("linkedinSearch") 
+              .setGoogleClientRequestInitializer(new CustomsearchRequestInitializer(KEY)) 
+              .build();
+
+    Customsearch.Cse.List list = cs.cse().list(currentBusiness.getName()).setCx(cx); 
+    List<Result> searchJsonResults = list.execute().getItems();
+    String[] numberOfFollowers;
+    int companyFollowers = 0;
+    if (searchJsonResults != null && searchJsonResults.size() != 0) {
+      Result linkedinBusiness = searchJsonResults.get(0);
+      String businessDescription = 
+        (String) linkedinBusiness.getPagemap().get("metatags").get(0).get("og:description");
+      if(businessDescription.contains(START_SUBSTRING) == true 
+          && businessDescription.contains(END_SUBSTRING) == true){
+        String followers = businessDescription.substring(
+                                businessDescription.indexOf(START_SUBSTRING) + 2, 
+                                businessDescription.indexOf(END_SUBSTRING) - 1);
+        try{
+          companyFollowers = Integer.parseInt(followers.replaceAll(",", ""));
+        } catch (NumberFormatException e){
+            LOGGER.warning(e.getMessage());
+        }
+      }
+      if (companyFollowers > MIN_FOLLOWERS) {
+        addBigBusinessToDatabase(currentBusiness);
+      } else {
+        checkNumberOfSimilarBusinessesInTheArea(currentBusiness,
+                                                similarBusinessesInTheArea);
+      }
+    }
+  }
+
+  private void checkNumberOfSimilarBusinessesInTheArea(Listing currentBusiness, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) {
+    int numberOfMatchingBusinesses = 0;
+    int i = 0;
+    while (i < similarBusinessesInTheArea.length 
+          && numberOfMatchingBusinesses < ALLOWED_NUMBER_OF_MATCHING_BUSINESSES) {
+        if (similarBusinessesInTheArea[i].name.contains(currentBusiness.getName())
```

Comment:
Just curious. Should this be contains or equals? I'm sure we can find cases that void either side, but it might be something you want to think about.

Reply:
I changed it to use the equals method for the time being, because on further thought it the results that are returned from the text search request should have the same name as the results are coming from the nearby search request results. I was scared that the name might differ a bit and would result in the business not being caught as a big. Nevertheless, I think it is the smart thing to do, unless I notice something later on.